sqlalchemy-stubs/sql/base.pyi

```diff
@@ -50,30 +52,32 @@ class SchemaEventTarget(object): ...
 class SchemaVisitor(ClauseVisitor):
     __traverse_options__: Any = ...
 
-class ColumnCollection(util.OrderedProperties[Column]):
-    def __init__(self, *columns: Column) -> None: ...
+class ColumnCollection(util.OrderedProperties[ColumnElement[Any]]):
```

Comment:
This PR gives a hundred false errors in our codebase likely because of this (or related) change. More precisely, now all `some_table.c.field` are inferred as `ColumnElement`, while in reality they are `Column` (a narrower type).

I have checked in the code, it looks like `ColumnElement`s are indeed allowed in this class. But the question is whether anything apart from `Column` is used in real code? If no, I would keep it `Column[Any]`, otherwise you can make `ColumnCollection` generic in `TypeVar('_C', bound=ColumnElement[Any])`, and update other annotations to use `ColumnCollection[Column[Any]]` (such as `FromClause.c`). Note: this is a dangerous thing, since you will need to update _all_ places where `ColumnCollection` or its subclasses are used.

Also I would add a test for something like `users.c.name`.

Reply:
I agree that it's unwise to make `ColumnCollection` generic and until we find a place where something other than `Column` is used, it should remain `Column[Any]`. I'll change it back to `Column[Any]` and add a test for `users.c.name`.